sdk/keyvault/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,141 @@
+
+```text
+Azure.RequestFailedException : Service request failed.
+Status: 403 (Forbidden)
+
+Content:
+{"error":{"code":"Forbidden","message":"Operation decrypt is not permitted on this key.","innererror":{"code":"KeyOperationForbidden"}}}
+```
+
+The operation and inner `code` may vary, but the rest of the text will indicate which operation is not permitted.
+This error indicates that the authenticated application nor users does not have permissions to perform that operation,
+though the cause may vary.
+
+1. Check that the application or user has the appropriate permissions:
+   * [Access policies](https://docs.microsoft.com/azure/key-vault/general/assign-access-policy) (Key Vault)
+   * [Role-Based Access Control (RBAC)](https://docs.microsoft.com/azure/key-vault/general/rbac-guide) (Key Vault and Managed HSM)
+2. If the appropriate permissions are assigned to your application or user, make sure you are authenticating as that user.
+   If using the [DefaultAzureCredential] a different credential might've been used than one you expected.
+   [Enable logging](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/README.md#logging)
+   and you will see which credential the [DefaultAzureCredential] used as shown below, and why previously-attempted credentials
+   were rejected.
+
+   ```text
+   [Informational] DefaultAzureCredential credential selected: Azure.Identity.EnvironmentCredential
+   ```
+
+#### Access Denied to First Party Service
+
+You may see an error similar to:
+
+```text
+Azure.RequestFailedException : Service request failed.
+Status: 403 (Forbidden)
+
+Content:
+{"error":{"code":"Forbidden","message":"Access denied to first party service. ...","innererror":{"code":"AccessDenied"}}}
+```
+
+The error `message` may also contain the tenant ID (`tid`) and application ID (`appid`). This error may occur because:
+
+1. You have the **Allow trust services** option enabled and are trying to access the Key Vault from a service not on
+   [this list](https://docs.microsoft.com/azure/key-vault/general/overview-vnet-service-endpoints#trusted-services) of
+   trust services.
+2. You are authenticated against a Microsoft Account (MSA) in Visual Studio or another credential provider. See
+   [above](#operation-not-permitted) for troubleshooting steps.
+
+## Other Service Errors
+
+To troubleshoot additional HTTP service errors not described below,
+visit [Azure Key Vault REST API Error Codes](https://docs.microsoft.com/azure/key-vault/general/rest-error-codes).
+
+### HTTP 429: Too Many Requests
+
+If you get an exception or see logs that describe HTTP 429, you may be making too many requests to Key Vault too quickly.
+
+Possible solutions include:
+
+1. Use a singleton for any `CertificateClient`, `KeyClient`, or `SecretClient` in your application for a single Key Vault.
+2. Use a single instance of [DefaultAzureCredential] for each Key Vault or Managed HSM endpoint you need to access.
```

Comment:
This can be misleading as a `DefaultAzureCredential` with no options configured will use a static instance that would be shared by any other clients.

Reply:
This is news to me, but I see from source now. Doesn't seem well advertised.

Reply:
It's not and we've considered changing this behavior since it can cause some unintuitive problems.

Reply:
I don't think we should advertise this behavior here either. I think that suggesting to use a single instance of a credential is good advice. Also I don't think we should narrow this to DefaultAzureCrededntial. Just use a single instance of the credential?

Reply:
I'll add "or other credential use to authenticate your clients". I do like advertising DAC in general, but can just drop its name here entirely if that's preferred (unresolve the comment if you want to see this changed).